plugins/modules/public_cloud_block_storage.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+
+from ansible.module_utils.basic import AnsibleModule
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: public_cloud_block_storage
+
+short_description: Manage OVH API for public cloud volume.
+
+description:
+    - This module manage, volume of an instance on OVH public Cloud.
+
+requirements:
+    - ovh >= 0.5.0
+
+options:
+    service_name:
+        required: true
+        description: The service_name
+    region:
+        required: true
+        description: The region where to deploy the volume
+    size:
+        required: true
+        description: Volume size (in GB)
+        type: integer
+    volume_type:
+        required: true
+        default: classic
+        description: Volume type
+        choices: [ 'classic', 'high-speed', 'high-speed-gen2']
+    name:
+        required: true
+        description: The Volume name
+    description:
+        required: false
+        description: Volume description
+    image_id:
+        required: false
+        description: The id of the image/os to deploy on the volume
+    snapshot_id:
+        required: false
+        description: snapshot id
+    state:
+        required: false
+        default: present
+        choices: ['present','absent']
+        description: Indicate the desired state of volume
+'''
+
+EXAMPLES = r'''
+- name: Ensure Volume is affected to instance
+  synthesio.ovh.public_cloud_block_storage:
+    service_name: "{{ service_name }}"
+    region: "{{ region }}"
+    size: "{{ size }}"
+    name: "volume-{{ inventory_hostname }}"
+    description: "Data Volume of {{ size }}"
+    volume_type: "{{ volume_type }}"
+  delegate_to: localhost
+  register: block_storage_metadata
+'''
+
+RETURN = r''' # '''
+
+from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import ovh_api_connect, ovh_argument_spec
+
+try:
+    from ovh.exceptions import APIError
+
+    HAS_OVH = True
+except ImportError:
+    HAS_OVH = False
+
+
+def run_module():
+    module_args = ovh_argument_spec()
+    module_args.update(dict(
+        service_name=dict(required=True),
+        region=dict(required=True),
+        size=dict(required=True, type="int"),
+        volume_type=dict(required=True, choices=['classic', 'high-speed', 'high-speed-gen2'], default='classic'),
+        name=dict(required=True),
+        description=dict(required=False),
+        image_id=dict(required=False),
+        snapshot_id=dict(required=False),
+        state=dict(choices=['present', 'absent'], default='present')
+    ))
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
```

Comment:
If you support `check_mode`, you should test if you are in check mode with `if module.check_mode:` (I know, this is not the case in all the plugins of this collections… ;))

Reply:
Done ;)